src/dc_receive_imf.rs

```diff
@@ -2632,4 +2622,119 @@ mod tests {
         // test that both messages got into deaddrop.
         assert!(!msg.chat_id.is_special());
     }
+
+    /// Test that classical MUA messages are assigned to group chats
+    /// based on the `In-Reply-To` header for two-member groups.
+    #[async_std::test]
+    async fn test_in_reply_to_two_member_group() {
+        let t = TestContext::new().await;
+        t.configure_addr("bob@example.com").await;
+
+        // Receive message from Alice about group "foo".
+        dc_receive_imf(
+            &t,
+            b"Received: (Postfix, from userid 1000); Mon, 4 Dec 2006 14:51:39 +0100 (CET)\n\
+                 From: alice@example.org\n\
+                 To: bob@example.com\n\
+                 Subject: foo\n\
+                 Message-ID: <message@example.org>\n\
+                 Chat-Version: 1.0\n\
+                 Chat-Group-ID: foo\n\
+                 Chat-Group-Name: foo\n\
+                 Date: Sun, 22 Mar 2020 22:37:57 +0000\n\
+                 \n\
+                 hello foo\n",
+            "INBOX",
+            1,
+            false,
+        )
+        .await
+        .unwrap();
+
+
+        // Receive a classic MUA reply from Alice.
+        // It is assigned to the group chat.
+        dc_receive_imf(
+            &t,
+            b"Received: (Postfix, from userid 1000); Mon, 4 Dec 2006 14:51:39 +0100 (CET)\n\
+                 From: alice@example.org\n\
+                 To: bob@example.com\n\
+                 Subject: Re: foo\n\
+                 Message-ID: <reply@example.org>\n\
+                 In-Reply-To: <message@example.org>\n\
+                 Date: Sun, 22 Mar 2020 22:37:57 +0000\n\
+                 \n\
+                 classic reply\n",
+            "INBOX",
+            2,
+            false,
+        )
+        .await
+        .unwrap();
+
+        // Ensure message is assigned to group chat.
+        let msg = t.get_last_msg().await;
+        let chat = Chat::load_from_db(&t, msg.chat_id).await.unwrap();
+        assert_eq!(chat.typ, Chattype::Group);
```

Comment:
i would also test that the group as acutally 2 messages, but this is maybe also a matter of taste :) nice test!